dynamiq/nodes/tools/exa_search.py

```diff
@@ -0,0 +1,177 @@
+from enum import Enum
+from typing import Any, ClassVar, Literal
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from dynamiq.connections import Exa, HTTPMethod
+from dynamiq.nodes import NodeGroup
+from dynamiq.nodes.node import ConnectionNode, ensure_config
+from dynamiq.runnables import RunnableConfig
+from dynamiq.utils.logger import logger
+
+
+class QueryType(str, Enum):
+    keyword = "keyword"
+    neural = "neural"
+    auto = "auto"
+
+
+class ExaInputSchema(BaseModel):
+    """Schema for Exa search input parameters."""
+
+    query: str = Field(description="The search query string.")
+    include_full_content: bool = Field(
+        default=False, description="If true, retrieve full content, highlights, and summaries for search results."
+    )
+    use_autoprompt: bool = Field(
+        default=False, description="If true, query will be converted to a Exa query. Default false."
+    )
+    query_type: QueryType = Field(
+        default=QueryType.auto,
+        description="Type of query to be used. Options are 'keyword', 'neural', or 'auto'. Default is 'auto'.",
+    )
+    category: str | None = Field(
+        default=None, description="A data category to focus on (e.g., company, research paper, news article)."
+    )
+    limit: int = Field(default=10, ge=1, le=100, description="Number of search results to return. Default 10.")
+    include_domains: list[str] | None = Field(default=None, description="List of domains to include in the search.")
+    exclude_domains: list[str] | None = Field(default=None, description="List of domains to exclude from the search.")
+    include_text: list[str] | None = Field(default=None, description="Strings that must be present in webpage text.")
+    exclude_text: list[str] | None = Field(
+        default=None, description="Strings that must not be present in webpage text."
+    )
+
+
+class ExaTool(ConnectionNode):
```

Comment:
why we need Tool in the name?

Reply:
Can use ‘ExaSearch’ instead of this

Reply:
Historically we are using Tool suffix for such nodes